lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -305,7 +305,8 @@ sub label {
 sub scenario {
     my ($self) = @_;
 
-    return $self->result_source->schema->resultset('TestSuites')->find({name => $self->TEST});
+    my $test_suite_name = $self->settings_hash->{TEST_SUITE_NAME} || $self->TEST;
+    return $self->result_source->schema->resultset('TestSuites')->find({name => $test_suite_name});
```

Comment:
I think this is a bit too UI-centric. Why not write the job template name into the TEST field when the job is *created*?

I think `$test_suite_name` can be misleading. There is no strong relation between TEST and the used test suite if any at all would have been used. E.g. if a job is just created with `openqa-client jobs post TEST=my_fancy_name` which does not relate to any test suite. Also there is the API to return a job and  there IMHO TEST should also reflect the final name. But as "name" is also already taken I do not have much ideas than `$test_suite_name`. Maybe `$template_name`?

Reply:
> I think this is a bit too UI-centric. Why not write the job template name into the TEST field when the job is created?
That's exactly what her code does and actually the reason why this modification in `sub scenario` is needed. Otherwise it wouldn't be possible to find the test suite anymore relying on `TEST`. Maybe that also explains your other question.

> I think $test_suite_name can be misleading. There is no strong relation between TEST and the used test suite if any at all would have been used.

But there is a relation which is now made more explicit (and the ` || $self->TEST` is only kept for backwards compatibility). And yes, this relation is only used to display the test suite description for the job so it is indeed not the "strongest" relation. However, I think the change makes still sense in order to keep things as functional as before.

Reply:
@Martchus I don't think `$self->TEST` is only kept for backwards compatibility. I don't think we ever set TEST_SUITE_NAME e.g. in case of `openqa-client jobs post`, right?

Reply:
Right, but in these cases it is also questionable whether TEST corresponds to a test suite at all (so I didn't consider it). Note that so far we also don't set the JOB_TEMPLATE_NAME.

Reply:
@okurz  `$self->TEST` is kept for backwards compatibility, and it's also used to deal with the situation that the job is created by `jobs post`. For `jobs post`, there is no `TEST_SUITE_NAME` and `JOB_TEMPLATE_NAME`, and in this context, the `TEST` is the test suite name.

@Martchus Thanks for your explanation.